src/main/java/com/example/booksstore/controller/BookController.java

```diff
@@ -17,32 +20,40 @@
 import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
 
+@Tag(name = "Book management", description = "Endpoints for managing books")
 @RequiredArgsConstructor
 @RestController
 @RequestMapping(value = "/api/books")
 public class BookController {
     private final BookService bookService;
 
+    @Operation(summary = "Get needed count of books from the shop",
+            description = "Get a list of all available books")
     @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.findAll();
+    public List<BookDto> getAll(Pageable pageable) {
+        return bookService.findAll(pageable);
     }
 
+    @Operation(summary = "Get book by id", description = "Gets a book from db")
     @GetMapping("/{id}")
     public BookDto getBookById(@PathVariable Long id) {
         return bookService.getById(id);
     }
 
+    @Operation(summary = "Create a book",
+            description = "Create a book with auto generated ID")
     @PostMapping
     public BookDto createBook(@RequestBody @Valid BookRequestDto bookDto) {
         return bookService.save(bookDto);
     }
 
+    @Operation(summary = "Updates a book", description = "Updates a book by ID")
     @PutMapping("/{id}")
     public BookDto updateBook(@PathVariable Long id, @RequestBody @Valid BookRequestDto bookDto) {
```

Comment:
Documenting some not-so-obvious fields of `BookRequestDto`  like ISBN maybe and its format that we accept, and that it should be unique - is all a very useful information for the whole team that we can add to the swagger, use `@Schema` above field for that. You may also check other fields as well, if you find it important

Reply:
I'm afraid you did not quite understand me right, what I meant is that we should use `@Schema` annotation and its attributes like `description`, `example` etc. above the `isbn` field (and maybe some other ones as well) in `BookRequestDto` to describe it there, and not in the endpoint itself, because if we have a list of 100 fields that we need to describe, doing it in the controller - is not the best idea